sites/geohub/src/lib/helper/getValueFromRasterTileUrl.ts

```diff
@@ -36,3 +36,31 @@ export const getValueFromRasterTileUrl = (
   }
   return value
 }
+
+export const getValueFromRasterTileUrlAsync = async (
+  map: Map,
+  layerId: string,
+  paramName: 'colormap_name' | 'rescale' | 'expression' | 'colormap' | 'url',
+): Promise<string | number[] | number[][][] | { [key: string]: number[] }> => {
+  await loadMap(map)
+  const source: RasterTileSource = map.getSource(map.getLayer(layerId).source) as RasterTileSource
+  if (source.type !== 'raster') {
+    throw new Error(`Invalid source type: ${source.type}`)
+  }
+  //if (!(source && source.tiles && source.tiles.length > 0)) await loadMap(map)
+  const tileUrl = new URL(source.tiles[0])
+  let value: string | number[] | number[][][] | { [key: string]: number[] } = tileUrl.searchParams.get(paramName)
+  if (value && paramName === 'rescale') {
+    const values = value.split(',')
+    value = values.map((v) => Number(v))
+  } else if (value && paramName === 'colormap') {
+    if (Array.isArray(value)) {
+      // interval
+      value = JSON.parse(value) as number[][][]
+    } else {
+      // unique
+      value = JSON.parse(value) as unknown as { [key: string]: number[] }
+    }
```

Comment:
Is it possible to call `getValueFromRasterTileUrl` function here? looks like same source code

Reply:
it is not exactly the same code, it is quite close. and unfortunately it is not possible to mix async and sync while preserting execution order. It is sad bu I had no choice